Lyx What you see is what you mean LaTeX editor

**Package description**
<!--
A short and clear description of what this package do and why it should
be packaged for Termux.
-->
What you see is what you mean LaTeX editor. Very powerful for quickly writing LaTeX documents. The main thing that's missing from Termux for me.

Its gui is QT and the other main dependency is LaTeX, both of which are already packaged. But there is probably still a handful of missing dependencies.

**Link to home page and sources**
1. Home page: https://www.lyx.org/
2. Source code: git://git.lyx.org/lyx or https://www.lyx.org/Download for tarballs of stable versions.

I tried copying the build.sh of qgit hoping that way the qt-dependency would already be resolved but couldn't make it work so far:
```
TERMUX_PKG_HOMEPAGE=https://www.lyx.org
TERMUX_PKG_DESCRIPTION="Lyx is a LaTeX editor"
TERMUX_PKG_LICENSE="GPL-2.0"
TERMUX_PKG_MAINTAINER=""
TERMUX_PKG_VERSION=2.3.3
TERMUX_PKG_REVISION=1
TERMUX_PKG_SRCURL=https://ftp.lip6.fr/pub/lyx/stable/2.3.x/lyx-2.3.3.tar.gz
TERMUX_PKG_SHA256=ec8027a0ee8d4d35d2d02e9c8edefb0e873e09347fbc11f6734cebca998d7d48
TERMUX_PKG_DEPENDS="qt5-base"
TERMUX_PKG_BUILD_IN_SRC=true

termux_step_configure() {
    export QTDIR="${TERMUX_PREFIX}/lib/cmake/Qt5"
    ./configure --host="${TERMUX_ARCH}" --enable-qt5 --prefix="${TERMUX_PREFIX}"
}
```
but this fails with

> configure: error: cannot compile a simple Qt executable. Check you have the right $QTDIR.

(using the provided docker image, on x86 Debian)

There another error related to qt:
```
g++ -o conftest  -I/data/data/com.termux/files/usr/include -I/data/data/com.termux/files/usr/include/Qt -I/data/data/com.termux/files/usr/include/QtCore -I/data/data/com.termux/files/usr/include/QtGui -I/data/data/com.termux/files/usr/include/QtWidgets -I/data/data/com.termux/files/usr/include/QtSvg -I/data/data/com.termux/files/usr/include/QtConcurrent -I/data/data/com.termux/files/usr/include/QtMacExtras -L/data/data/com.termux/files/usr/lib     conftest.cpp  -lSM -lICE   -lX11  -lQt5Widgets >&5
/data/data/com.termux/files/usr/bin/aarch64-linux-android-ld: /data/data/com.termux/files/usr/tmp/conftest-b9e59a.o: undefined reference to symbol '_ZN10QArrayData10deallocateEPS_mm@@Qt_5'
/data/data/com.termux/files/usr/bin/aarch64-linux-android-ld: /data/data/com.termux/files/usr/lib/libQt5Core.so.5: error adding symbols: DSO missing from command line
clang-8: error: linker command failed with exit code 1 (use -v to see invocation)
```

--- a/packages/lyx/build.sh
+++ b/packages/lyx/build.sh
@@ -1,0 +1,17 @@
+TERMUX_PKG_HOMEPAGE=http://www.lyx.org
+TERMUX_PKG_DESCRIPTION="WYSIWYM (What You See Is What You Mean) Document Processor"
+TERMUX_PKG_LICENSE="GPL-2.0"
+TERMUX_PKG_MAINTAINER="Simeon Huang <symeon@librehat.com>"
+TERMUX_PKG_VERSION=2.3.6.1
+TERMUX_PKG_SRCURL="https://ftp.lip6.fr/pub/lyx/stable/2.3.x/lyx-${TERMUX_PKG_VERSION}.tar.xz"
+TERMUX_PKG_SHA256=c6bed2633419898e01a7cc350310d7d934bf591cab7259fc2581aa4c00eafa78
+TERMUX_PKG_DEPENDS="qt5-qtbase, qt5-qtsvg, zlib, imagemagick, ghostscript, texlive-bin, hunspell"
+TERMUX_PKG_BUILD_DEPENDS="autoconf, automake, bc, python, boost, qt5-qtbase-cross-tools"
+TERMUX_PKG_EXTRA_CONFIGURE_ARGS="
+--enable-build-type=rel
+--enable-qt5
+--without-included-boost
+--without-aspell
+--with-hunspell
+"
+TERMUX_PKG_RM_AFTER_INSTALL="share/lyx/examples"

--- a/packages/lyx/build.sh
+++ b/packages/lyx/build.sh
@@ -15,3 +15,9 @@ TERMUX_PKG_EXTRA_CONFIGURE_ARGS="
 --with-hunspell
 "
 TERMUX_PKG_RM_AFTER_INSTALL="share/lyx/examples"
+
+
+termux_step_pre_configure() {
+    # This is to allow the build script find the `moc` on cross-build host
+    export PATH+=":${TERMUX_PREFIX}/opt/qt/cross/bin"
+}

--- a/packages/lyx/build.sh
+++ b/packages/lyx/build.sh
@@ -6,7 +6,7 @@ TERMUX_PKG_VERSION=2.3.6.1
 TERMUX_PKG_SRCURL="https://ftp.lip6.fr/pub/lyx/stable/2.3.x/lyx-${TERMUX_PKG_VERSION}.tar.xz"
 TERMUX_PKG_SHA256=c6bed2633419898e01a7cc350310d7d934bf591cab7259fc2581aa4c00eafa78
 TERMUX_PKG_DEPENDS="qt5-qtbase, qt5-qtsvg, zlib, imagemagick, ghostscript, texlive-bin, hunspell"
-TERMUX_PKG_BUILD_DEPENDS="autoconf, automake, bc, python, boost, qt5-qtbase-cross-tools"
+TERMUX_PKG_BUILD_DEPENDS="boost, qt5-qtbase-cross-tools"
 TERMUX_PKG_EXTRA_CONFIGURE_ARGS="
 --enable-build-type=rel
 --enable-qt5

